components/event-sources/adapter/http/adapter.go

```diff
@@ -0,0 +1,196 @@
+package http
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/kyma-project/kyma/components/event-sources/apis/sources"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/adapter"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/source"
+	"log"
+	"net/http"
+)
+
+type envConfig struct {
+	adapter.EnvConfig
+	EventSource string `envconfig:"EVENT_SOURCE" required:"true"`
+
+	// PORT as required by knative serving runtime contract
+	Port int `envconfig:"PORT" required:"true" default:"8080"`
```

Comment:
Do these get populated without calling `envconfig.Process()`? If yes, how does this black magic work?